RecoBTag/TensorFlow/python/pfDeepDoubleCvLJetTags_cfi.py

```diff
@@ -0,0 +1,21 @@
+import FWCore.ParameterSet.Config as cms
+
+pfDeepDoubleCvLJetTags = cms.EDProducer('DeepDoubleXTFJetTagsProducer',
+  src = cms.InputTag('pfDeepDoubleXTagInfos'),
+  input_names = cms.vstring(
+    'input_1',
+    'input_2',
+    'input_3'
+  ),
+  graph_path = cms.FileInPath('RecoBTag/Combined/data/DeepDoubleX/94X/V01/DDC.pb'),
+  lp_names = cms.vstring('db_input_batchnorm/keras_learning_phase'),
```

Comment:
in deepFlavour a similar string starts with ```cpf_input_batchnorm```.
what is the difference between CPF and DB?

Reply:
This has to do with how the model is saved in the protobuf file. TensorFlow has to find the boolean tensor which stores whether the model will be used in inference model (False) or training mode (True). The behavior of the batch normalization layers is different in the two modes.

In our protobuf file, this boolean happens to be called `db_input_batchnorm` (because our first set of inputs is the dedicated double-b inputs), whereas in the DeepFlavour protobuf, this boolean is called `cpf_input_batchnorm` (because their first set of inputs are the charged PF candidate inputs). So in general, it's good to have this configurable.

Reply:
db = doubleb and cpf = charge pf.
This makes sense now. Thanks.

Am I correct to assume that we are using the inference model here?

Reply:
Yes, we are using the inference mode here. So we "freeze" the graph to make the variables constant (and set that boolean to False) before we save it in the protobuf file.